src/components/AppIndex.vue

```diff
@@ -39,34 +60,126 @@ export default {
+                return {
+                    'name': kw,
+                    'count': $this.rawKeywords.filter((kw2) => kw === kw2).length
+                }
+            }).sort(function (k1, k2) {
+                if(k1.count != k2.count) {
+                    // Distinguish by number
+                    return k2.count - k1.count
+                }
+                // Distinguish by keyword name
+                return (k1.name.toLowerCase() > k2.name.toLowerCase()) ? 1 : -1
+            })
+        },
+        // An array of keyword objects that are currently in use for filtering
+        selectedKeywords() {
+            return this.keywordsWithCount.filter((kw) => this.keywordFilter.includes(kw.name))
+        },
+        // An array of those keyword objects that are currently not in use for filtering
+        unselectedKeywords() {
+            return this.keywordsWithCount.filter((kw) => ! this.selectedKeywords.includes(kw))
+        },
+        // An array of all recipes that are part in all filtered keywords
+        recipesFilteredByKeywords() {
+            let $this = this
+            return this.recipes.filter(function (r) {
+                if($this.keywordFilter.length == 0) {
+                    // No filtering, keep all
+                    return true
+                }
+                
+                if(r.keywords === null){
+                    return false
+                }
+                
+                function keywordInRecipePresent(kw,r) {
+                    if(! r.keywords) {
+                        return false;
                     }
+                    
+                    let keywords = r.keywords.split(',')
+                    return keywords.includes(kw.name)
                 }
-                return filtered
+                
+                return $this.selectedKeywords.map((kw) => keywordInRecipePresent(kw, r)).reduce((l,r) => l && r)
+            })
+        },
+        // An array of the finally filtered recipes, that is both filtered for keywords as well as string-based name filtering
+        filteredRecipes () {
+            let ret = this.recipesFilteredByKeywords
+            let $this = this
+            
+            if(this.filters){
+                ret = ret.filter(function (r) {
+                    return r.name.toLowerCase().includes($this.filters.toLowerCase())
+                })
             }
+            
+            return ret
+        },
+        // An array of keywords that are yet unselected but some visible recipes are associated
+        selectableKeywords() {
+            if (this.unselectedKeywords.length === 0) {
+                return []
+            }
+            
+            let $this = this
+            return this.unselectedKeywords.filter(function (kw) {
+                return $this.filteredRecipes.map(function (r) {
+                    return r.keywords && r.keywords.split(',').includes(kw.name)
+                }).reduce((l,r) => l || r, false)
```

Comment:
Do you know if this is handled in such a clever way that it already returns if it hits the first `true`?

Reply:
I do not think so. It first generates the whole bool array and then merges it. The benefit is that the array operation can be parallalized that way (in theory).